cs/docs/integrace-do-aplikace.md

```diff
@@ -0,0 +1,114 @@
+
+
+## Vytvoření připojení  {#toc-pripojeni}
+
+Vytvoříme si objekt `LeanMapper\Connection` a předáme mu parametry potřebné pro připojení k databázi.
+
+``` php?start_inline=1
+$connection = new LeanMapper\Connection(array(
+    'driver'   => 'mysqli',
+    'host'     => 'localhost',
+    'username' => 'root',
+    'password' => '***',
+    'database' => 'mydatabase',
+));
+```
+
+**Tip:** `LeanMapper\Connection` přebírá stejné parametry jako třída [`DibiConnection`](https://api.dibiphp.com/3.0/Dibi.Connection.html).
+
+
+## Mapper  {#toc-mapper}
+
+Dále potřebujeme objekt implementující [`LeanMapper\IMapper`](/cs/docs/mapper/). Lean Mapper nám poskytuje výchozí implementaci formou třídy `LeanMapper\DefaultMapper`.
+
+``` php?start_inline=1
+$mapper = new LeanMapper\DefaultMapper;
+```
+
+
+## Entity factory  {#toc-entityfactory}
+
+A jako poslední věc vytvoříme objekt, který implementuje rozhraní [`LeanMapper\IEntityFactory`](/cs/docs/entity-factory). Lean Mapper nám opět podává pomocnou ruku prostřednictvím třídy `LeanMapper\DefaultEntityFactory`.
+
+``` php?start_inline=1
+$entityFactory = new LeanMapper\DefaultEntityFactory;
+```
+
+
+## Repositáře  {#toc-repositare}
+
+Abychom mohli pracovat s [entitami](/cs/docs/entity/), potřebujeme k tomu [repositáře](/cs/docs/repositare/). Předpokládejme, že máme repositář napsaný, nyní ho jen vytvoříme a předáme mu potřebné závislosti.
+
+``` php?start_inline=1
+$bookRepository = new Model\BookRepository($connection, $mapper, $entityFactory);
+```
+
+A to je vše.
+
+
+## Integrace do Nette aplikace  {#toc-nette}
+
+Ještě si ukážeme, jak použít Lean Mapper v rámci aplikace napsané v [Nette](https://nette.org). Do konfiguračního souboru aplikace (`config.neon`) si přídáme následující parametry a definice služeb:
+
+``` yaml
+parameters:
+    # údaje pro připojení k DB
+    leanmapper:
+        driver: mysqli
+        host: localhost
+        username: ...
+        password: ...
+        database: mydatabase
+
+# ...
+
+services:
+    # registrace Lean Mapperu
+    - LeanMapper\Connection(%leanmapper%)
+    - LeanMapper\DefaultMapper
+    - LeanMapper\DefaultEntityFactory
+
+    # registrace repositářů
+    - Model\BookRepository
+    - Model\AuthorRepository
+    - ...
+
+    # ...
+```
+
+Repositářům nemusíme ručně předávat závislosti, o to se automaticky postará [auto-wiring](http://doc.nette.org/cs/2.4/configuring#toc-auto-wiring) v Nette.
+
+
+### Předání repositáře do presenteru  {#toc-nette-presenter}
+
+Pro předání repositáře do presenteru můžeme využít anotaci `@inject`.
+
+``` php?start_inline=1
+class BookPresenter extends BasePresenter {
+    /** @var \Model\BookRepository @inject */
+    public $bookRepository;
+
+    ...
+}
+```
+
+V rámci presenteru pak máme repositář přístupný přes `$this->bookRepository`.
+
+**Tip:** pokud nechcete, nebo nemůžete použít anotaci `@inject`, lze pro předání repositáře použít constructor nebo [`inject*()` metodu](http://pla.nette.org/cs/inject-autowire).
```

Comment:
Jasně, že ano. Psal jsem to hlavně s ohledem na začátečníky, kteří se snadno do contructor hellu v presenteru zapletou a jeho vyřešení je pro ně obtížné, pokročilejší uživatelé tohle nepotřebují, ti si to rovnou předají přes kontruktor.